interval/search_tree.go

```diff
@@ -178,3 +180,35 @@ func (st *MultiValueSearchTree[V, T]) IsEmpty() bool {
 
 	return st.root == nil
 }
+
+// GobEncode encodes the tree (compatible with encoding/gob).
+func (st *MultiValueSearchTree[V, T]) GobEncode() ([]byte, error) {
+	var b bytes.Buffer
+	enc := gob.NewEncoder(&b)
+
+	if err := enc.Encode(st.root); err != nil {
+		return nil, err
+	}
+
+	if err := enc.Encode(st.config.allowIntervalPoint); err != nil {
+		return nil, err
+	}
+
+	return b.Bytes(), nil
+}
+
+// GobDecode decodes the tree (compatible with encoding/gob).
+func (st *MultiValueSearchTree[V, T]) GobDecode(data []byte) error {
+	b := bytes.NewBuffer(data)
+	enc := gob.NewDecoder(b)
+
+	if err := enc.Decode(&st.root); err != nil {
+		return err
+	}
+
+	if err := enc.Decode(&st.config.allowIntervalPoint); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
What do you think of implementing this to the `SearchTree` type too? So we have both public interval tree implementing the encoding/gob interfaces.

If you agree with that, please add tests to the `SearchTree` type too, as you have done for the `MultiValueSearchTree`.

Reply:
Done: [Add encoding/decoding support for SearchTree](https://github.com/rdleal/intervalst/pull/9/commits/7671d471007d91e38afceb298dc52fdb0bad984a)